pkg/controller/logstash/keystore.go

```diff
@@ -22,10 +22,53 @@ const (
 )
 
 var (
-	keystoreCommand          = "echo 'y' | /usr/share/logstash/bin/logstash-keystore"
+	// containerCommand runs in every pod creation to regenerate keystore.
+	// `logstash-keystore` allows for adding multiple keys in a single operation.
+	// All keys and values must be ASCII and non-empty string. Values are input via stdin, delimited by \n.
+	containerCommand = `#!/usr/bin/env bash
+
+set -eu
+
+{{ if not .SkipInitializedFlag -}}
+keystore_initialized_flag={{ .KeystoreVolumePath }}/elastic-internal-init-keystore.ok
+
+if [[ -f "${keystore_initialized_flag}" ]]; then
+    echo "Keystore already initialized."
+	exit 0
+fi
+
+{{ end -}}
+echo "Initializing keystore."
+
+# create a keystore in the default data path
+{{ .KeystoreCreateCommand }}
+
+# add all existing secret entries to keys (Array), vals (String). 
+for filename in  {{ .SecureSettingsVolumeMountPath }}/*; do
+	[[ -e "$filename" ]] || continue # glob does not match
+	key=$(basename "$filename")
+	keys+=("$key")
+	vals+=$(cat "$filename")
+	vals+="\n"
+done
+
+# remove the trailing '\n' from the end of the vals
+vals=${vals%'\n'}
+
+# add multiple keys to keystore
+{{ .KeystoreAddCommand }}
+
+{{ if not .SkipInitializedFlag -}}
+touch {{ .KeystoreVolumePath }}/elastic-internal-init-keystore.ok
+{{ end -}}
+
+echo "Keystore initialization successful."
+`
+
 	initContainersParameters = keystore.InitContainerParameters{
-		KeystoreCreateCommand:         keystoreCommand + " create",
-		KeystoreAddCommand:            keystoreCommand + ` add "$key" < "$filename"`,
+		KeystoreCreateCommand:         "echo 'y' | /usr/share/logstash/bin/logstash-keystore create",
+		KeystoreAddCommand:            `echo -e "$vals" | bin/logstash-keystore add "${keys[@]}"`,
```

Comment:
```suggestion
		KeystoreAddCommand:            `echo -e "$vals" | /usr/share/logstash/bin/logstash-keystore add "${keys[@]}"`,
```

Nit: Inconsistent invocation of `logstash-keystore`